nuxeo-java-client/src/main/java/org/nuxeo/client/objects/Repository.java

```diff
@@ -29,293 +27,426 @@
 import org.nuxeo.client.objects.workflow.Workflow;
 import org.nuxeo.client.objects.workflow.Workflows;
 
+import okhttp3.ResponseBody;
 import retrofit2.Callback;
 
 /**
  * @since 0.1
  */
 public class Repository extends RepositoryEntity<RepositoryAPI> {
 
-    protected Document root;
-
     public Repository(NuxeoClient nuxeoClient) {
         super(EntityTypes.DOCUMENT, RepositoryAPI.class, nuxeoClient);
     }
 
-    public Repository repositoryName(String repositoryName) {
-        super.repositoryName = repositoryName;
-        return this;
+    public Repository(NuxeoClient nuxeoClient, String repositoryName) {
+        this(nuxeoClient);
+        this.repositoryName = repositoryName;
     }
 
-    public Document getDocumentRoot() {
-        return root;
+    public Document fetchDocumentRoot() {
+        if (repositoryName == null) {
+            return fetchResponse(api.fetchDocumentRoot());
+        }
+        return fetchResponse(api.fetchDocumentRoot(repositoryName));
+    }
+
+    public void fetchDocumentRoot(Callback<Document> callback) {
+        if (repositoryName == null) {
+            fetchResponse(api.fetchDocumentRoot(), callback);
+        } else {
+            fetchResponse(api.fetchDocumentRoot(repositoryName), callback);
+        }
     }
 
     /* By Id - Sync */
 
     public Document fetchDocumentById(String documentId) {
-        return (Document) getResponse(documentId);
+        if (repositoryName == null) {
+            return fetchResponse(api.fetchDocumentById(documentId));
+        }
+        return fetchResponse(api.fetchDocumentById(documentId, repositoryName));
```

Comment:
Can't the `api.fetchDocumentById(documentId, repositoryName)` deal with the `repositoryName == null` case itself? This seems to be a lot of repetition in all similar methods.